legal-api/src/legal_api/services/warnings/business/business_checks/firms.py

```diff
@@ -257,7 +257,8 @@ def check_address(address: Address,
                                                    address_type,
                                                    BusinessWarnings.NO_ADDRESS_POSTAL_CODE))
 
-    if referer != BusinessWarningReferers.BUSINESS_OFFICE and not address.region:
+    if referer not in (BusinessWarningReferers.BUSINESS_OFFICE, BusinessWarningReferers.BUSINESS_PARTY,
+                       BusinessWarningReferers.COMPLETING_PARTY) and not address.region:
```

Comment:
I think it should be only adding a no address region warning if all 3 conditions below are met.  This would be the case where we expect the business office address to always have a non-null region value.  i.e. it should allways have a value of "BC"

1. referer == BusinessWarningReferers.BUSINESS_OFFICE
2. address_type == 'Address.DELIVERY
3. address.region is null

Reply:
Makes way more sense to me now, thank you